### DISCUSSION TITLE
Exclusive access via touch - possible?

### DISCUSSION THREAD

**Discussion by claudiusraphael** [Schneegans/Fly-Pie#271] [Ideas & Feature Discussions]
I wonder if it would be possible to offer an exclusive touch-mode for the menu, meaning preventing fly-pie being used with other input-devices.

For example on EliteBooks and Cintiqs with Wacom Pen and Touch when using drawing applications to use the pen (accidentally or intentional) can be disruptive, whereas touch-input would be perfect, as touch even with gnomes gestures in use feels natural.

A touch-optimized mode would be very much appreciated, for example adding a cancel-button (because a long-press to right-click/cancel is disruptive as well) or alternatively if that is technically possible even overwrite gnome/application-touch-input-processing temporarily while the menu is open and add gestures to handle the menu.

**Comment by Schneegans** [comment]
Hi there; thanks for the suggestion! However, I think that I do not really understand what you mean. Could you try to explain in more detail? What would happen in the "exclusive touch-mode"? I get the idea of the "cancel button" (#260). Yet what kind of gestures would you like to use to "handle the menu"?

**Reply by claudiusraphael** [reply]
Hmm, let me try:

A main problem on all platforms is that either (if you are lucky) the operating system defines gestures like two/three/four-finger-swipes and -rotations, pinches/clutches systemwide and/or the operating system allows overrides and applications do their own thing (which can be helpful for that specific app, but since there is no real standard ever dev/team does its own "thing").

To overcome that i tried to use pen-touch-screens/tablets either directly on linux desktops and specifiy which input device shall be accessible to what application (its a pita and depends on lot of nifty on abuse bordering rerouting of the input devices) or to make it a cross-platform-approach and generalize the behavior i run the apps either remotely via x11 (works for touch) or for drawing tools which offer pen-pressure make the pen exclusive to the app by running it in a VM.

This is all a pain.

What i wonder is if fly-pie could take control of specific input-devices and override/disable os- and app- specific handling for touch- and pen- gestures.

As you can see, yes - i have a dream - but .. yknow, no wordz.

So as a more proper example:

- imitating the generally available gestures given on a apple-device, like the four-finger up-/down-swipes as shortcut to show active app windows on current desktop / show mission control
... but reflected in fly-pie, for example by switching to another ring that reflects the shown apps, minimized apps, active desktops/spaces and similar

- using multi-touch four-finger/two-handed gestures aka two fingers on each hand performing the gesture or if you wanna flex do the spock-"nano nano"-wink alternatively
... for example to switch to another type/perspective/category of fly-pie - kinda like switching desktops or in an editor switching between modal and direct input or like switching windowing-mode between float and boxed

I guess i could describe in detail example-modes if i know what would be possible technically .. maybe?

Anyway, thanks for being curious about it and sorry for making not much sense, lol!

**Reply by claudiusraphael** [reply]
Oh and regarding the "preventing" fly-pie being used with other input devices:

Assuming it is possible to control specific input-devices it should be possible to completely pass-through specific input-devices to specific apps, right?

Basically i think pie-menus are still one of the most superior menus out there if they have control over everything and allow templating as doing custom configurations in each and any app and os is the most massive headache one can choose.

So having fly-pie being the inter-mediator - representing visually what mode and context one is in - makes it much more intuitive than flying blind through UI's trying to grasp and memorize all the possibilities touchpad, trackpad, touchscreen and pen-input offer. Or if you are present- and future- aware AR/VR-controls as well as accessibility-improving controllers for disabled people.

If you can please consider this "request" as a spot to provide what might be possible in that regard (even if it is not yet possible).

Thanks for taking the time and for keeping this project running in general, much appreciated!

**Reply by claudiusraphael** [reply]
Btw. i am aware of [the response on issue 221](https://github.com/Schneegans/Fly-Pie/issues/221#issuecomment-1092088979), but am not aware what the actual status for fly-pie specifically is - if something changed internally or on Gnome's end - that would as of now offer different possibilities.
